scripts/bert/bertpass_gpu.cc

```diff
@@ -218,229 +219,277 @@ class Graph {
-          g.nodes.push_back(node_expand_2_bias);
-          
-          // create broadcast_like node
-          Node* node_bcst_like = new Node();
-          node_bcst_like->name = base_name + "_broadcast_like";
-          node_bcst_like->op = "broadcast_like";;
-          node_bcst_like->inputs.resize(2);
-          node_bcst_like->inputs[0].node = node_expand_2_bias;
-          node_bcst_like->inputs[0].entry = 0;
-          node_bcst_like->inputs[1].node = node_ffn1_fwd;
-          node_bcst_like->inputs[1].entry = 0;
-          g.nodes.push_back(node_bcst_like);
-          
-          // create BiasAdd Node
-          Node* node_add_bias = new Node();
-          node_add_bias->name = base_name + "_add_bias";
-          node_add_bias->op = "elemwise_add";
-          node_add_bias->inputs.resize(2);
-          node_add_bias->inputs[0].node = node_ffn1_fwd;
-          node_add_bias->inputs[0].entry = 0;
-          node_add_bias->inputs[1].node = node_bcst_like;
-          node_add_bias->inputs[1].entry = 0;
-          g.nodes.push_back(node_add_bias);
-          
-          //set BiasAdd node as gelu input
-          node_gelu->inputs[0].node = node_add_bias;
-          node_gelu->inputs[0].entry = 0;
-      }    
+    std::string str_ffn1 = "ffn_1_fwd";
+    if (n->name.find(str_ffn1) != std::string::npos) {
+      Node* node_ffn1_fwd = n;
+      Node* node_ffn1_bias = node_ffn1_fwd->inputs[2].node;
+      Node* node_gelu = node_ffn1_fwd->outputs[0].node;
+
+      std::size_t pos = n->name.find("fwd");
+      std::string base_name = n->name.substr(0, pos - 1);
+
+      // remove Bias terms in FC
+      node_ffn1_fwd->attrs["no_bias"]="True";
+      node_ffn1_fwd->inputs.pop_back();
+
+      // create 4 new nodes: 2 expand_dims nodes to expand bias dimensions
+      // a broadcast_like node, and BiasAdd
+#if MXNET_1_7
+      Node* node_expand_1_bias = new Node();
+      node_expand_1_bias->name = base_name + "_expand_1_bias";
+      node_expand_1_bias->op = "expand_dims";
+      Node* node_expand_2_bias = new Node();
+      node_expand_2_bias->name = base_name + "_expand_2_bias";
+      node_expand_2_bias->op = "expand_dims";
+      Node* node_bcst_like = new Node();
+      node_bcst_like->name = base_name + "_broadcast_like";
+      node_bcst_like->op = "broadcast_like";
+      Node* node_add_bias = new Node();
+      node_add_bias->name = base_name + "_add_bias";
+      node_add_bias->op = "elemwise_add";
+      g->nodes.push_back(node_expand_1_bias);
+      g->nodes.push_back(node_expand_2_bias);
+      g->nodes.push_back(node_bcst_like);
+      g->nodes.push_back(node_add_bias);
+#else
+      Node* node_expand_1_bias = g->addNode(base_name + "_expand_1_bias", "expand_dims");
+      Node* node_expand_2_bias = g->addNode(base_name + "_expand_2_bias", "expand_dims");
+      Node* node_bcst_like = g->addNode(base_name + "_broadcast_like", "broadcast_like");
+      Node* node_add_bias = g->addNode(base_name + "_add_bias", "elemwise_add");
```

Comment:
Its nice to see that from 1.7 --> 1.8 we were able to condense the number of lines of code users need to write from 16 down to 4! 👍